lib/galaxy/webapps/galaxy/api/object_store.py

```diff
@@ -0,0 +1,80 @@
+"""
+API operations on Galaxy's object store.
+"""
+import logging
+from typing import (
+    Any,
+    Dict,
+    List,
+)
+
+from fastapi import (
+    Path,
+    Query,
+)
+
+from galaxy.exceptions import (
+    ObjectNotFound,
+    RequestParameterInvalidException,
+)
+from galaxy.managers.context import ProvidesUserContext
+from galaxy.objectstore import BaseObjectStore
+from . import (
+    depends,
+    DependsOnTrans,
+    Router,
+)
+
+log = logging.getLogger(__name__)
+
+router = Router(tags=["object sstore"])
+
+ConcreteObjectStoreIdPathParam: str = Path(
+    ..., title="Concrete Object Store ID", description="The concrete object store ID."
+)
+
+SelectableQueryParam: bool = Query(
+    default=False, title="Selectable", description="Restrict index query to user selectable object stores."
+)
+
+
+@router.cbv
+class FastAPIObjectStore:
+    object_store: BaseObjectStore = depends(BaseObjectStore)
+
+    @router.get(
+        "/api/object_store",
```

Comment:
```suggestion
        "/api/object_stores",
```

it seems like most (every?) collection in our API is in plural. I think that's a good convention. Is there a reason you're deviating from this ?